lib/usercheck_integration.rb

```diff
@@ -0,0 +1,130 @@
+##
+# UserCheck.com Integration for Alaveteli UserSpamScorer
+# This module provides integration with the UserCheck.com API for email domain
+# validation
+#
+# Usage:
+#   1. Configure your UserCheck.com API key in your application settings
+#   2. Require this file in your theme initialization
+#   3. The integration will automatically register custom scoring methods
+#
+# Configuration:
+#   Set USERCHECK_API_KEY environment variable
+module UserCheckIntegration
+  API_BASE_URL = 'https://api.usercheck.com'
+  REQUEST_TIMEOUT = 5.seconds
+  CACHE_DURATION = 28.days
+
+  class << self
+    def api_key
+      @api_key ||= ENV['USERCHECK_API_KEY']
+    end
+
+    attr_writer :api_key
+
+    def enabled?
+      api_key.present?
+    end
+
+    def check_domain(domain)
+      return { valid: true, disposable: false, risky: false } unless enabled?
+
+      cache_key = "usercheck_domain_#{domain}"
+      cached_result = Rails.cache.read(cache_key)
+      return cached_result if cached_result
+
+      result = make_api_request(domain)
+
+      if result[:success]
+        Rails.cache.write(cache_key, result, expires_in: CACHE_DURATION)
+      end
+
+      result
+    end
+
+    private
+
+    def make_api_request(domain)
+      uri = URI("#{API_BASE_URL}/domain/#{domain}")
+
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = true
+      http.read_timeout = REQUEST_TIMEOUT
+      http.open_timeout = REQUEST_TIMEOUT
+
+      request = Net::HTTP::Get.new(uri)
+      request['Authorization'] = "Bearer #{api_key}"
+      request['Content-Type'] = 'application/json'
+
+      response = http.request(request)
+
+      case response.code
+      when '200'
+        data = JSON.parse(response.body)
+        {
+          success: true,
+          valid: data['valid'] != false,
```

Comment:
valid isn't in the API, looking at https://www.usercheck.com/docs/api/domain-endpoint